### ISSUE TITLE
Provide a flexible and simple mechanism to filter entries from the HAR

### ISSUE THREAD

**Issue by derevnjuk** [NeuraLegion/cypress-har-generator#163]
**Description**
It would be useful to have a way to filter recorded entries from the resulting HAR file, as the current options (i.e. `cy.recordHar({ /* ... */})`) do not provide enough flexibility for certain use cases.

**Possible solution**
We can add a query parameter that allows the user to filter the recorded entries using a Lucene-like syntax, XPath, or JSONPath. For example: 

```ts
cy.recordHar({ query: 'url:*example.com/* AND contentLength:[10 TO 100]' });
```
or
```ts
cy.recordHar({ query: `$[?(@.url.includes('example.com') && @.contentLength >= 10 && @.contentLength <= 100)]` });
```
Alternatively, we can allow specifying the path to the module being used to filter them as follows:

```js
// filter.js
module.exports = request => {
  return /example\.com/.test(request.url) && request.contentLength < 100 && request.contentLength > 10;
}

// in the test
cy.recordHar({
  filter: "path/to/filter.js"
});
```

**Additional context**
Cypress is spawning an independent child process that runs a plugin. The test is actually running in the browser. While communicating with each other, Cypress serializes command arguments and passes them to the plugin via WebSocket. Therefore, passing a plain JS function is hardly possible. This feature will give the users more control over the data that is saved and make it more efficient.

**Comment by derevnjuk** [comment]
@cheapsteak I am pleased to inform you that the feature you have been interested in (https://github.com/NeuraLegion/cypress-har-generator/issues/85#issuecomment-1377521058) has been released in version 5.13.0.

**Comment by cheapsteak** [comment]
Thank you so much!  
![image](https://user-images.githubusercontent.com/743976/214382796-f17e9022-b8de-4da7-91d1-6970b8bf16ce.png)
